File: week1/cpp/C.cpp
#include <bits/stdc++.h>
using namespace std;

// Function to check if a number is prime.
bool isprime(long long n) {
  for (long long i = 2; i * i <= n; i++) {
    if (n % i == 0)
      return false; // If 'n' is divisible by any number between 2 and sqrt(n), it's not prime.
  }
  return true; // If no divisors are found, 'n' is prime.
}

int main() {
  long long n;
  cin >> n; // Read the number 'n' from the standard input.

  if (isprime(n))
    cout << "Yes\n"; // If 'n' is prime, print "Yes".
  else
    cout << "No\n"; // If 'n' is not prime, print "No".
}

File: week1/py/C.py
import math

# Read the integer 'n' from the user.
n = int(input())

# Initialize a flag variable to 0.
flag = 0

# Iterate through potential divisors from 2 to the square root of 'n'.
for i in range(2, int(math.sqrt(n)) + 1):
    # Check if 'n' is divisible by 'i'.
    if n % i == 0:
        # If a divisor is found, set 'flag' to 1 and break out of the loop.
        flag = 1
        break

# Check the value of 'flag' to determine if 'n' is prime.
if flag == 1:
    print("No")  # 'n' is not prime
else:
    print("Yes") # 'n' is prime
